### PR TITLE
doc changes

### PR DESCRIPTION
**Pull request by margaretmeehan**
Crucial updates to developer versus user guide. This is still a WIP but would like to get these changes in ASAP.

### All Submissions:

- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Microsoft/agogosml/blob/master/CONTRIBUTING.rst) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Microsoft/agogosml/pulls) for the same update/change?
- [ ] Does your PR follow our [Code of Conduct](https://opensource.microsoft.com/codeofconduct/)?

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Does each method or function "do one thing well"? Reviewers may recommend methods be split up for maintainability and testability.
- [ ] Is this code designed to be testable? 
- [ ] Is the code documented well?
- [ ] Does your submission pass existing tests (or update existing tests with documentation regarding the change)?
- [ ] Have you added tests to cover your changes?
- [ ] Have you linted your code prior to submission?
- [ ] Have you updated the documentation and README?
- [ ] Is PII treated correctly? In particular, make sure the code is not logging objects or strings that might contain PII (e.g. request headers). 
- [ ] Have secrets been stripped before committing?

### CODE REVIEW

File: agogosml_cli/cli/templates/{{cookiecutter.PROJECT_NAME_SLUG}}/dockerbuild.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-docker build --no-cache -t agogosml -f agogosml/Dockerfile.agogosml agogosml/
-docker build --no-cache -t input_reader -f input_reader/Dockerfile.input_reader input_reader/
-docker build --no-cache -t output_writer -f output_writer/Dockerfile.output_writer output_writer/
-docker build --no-cache -t sample_app -f {{cookiecutter.PROJECT_NAME_SLUG}}/Dockerfile.{{cookiecutter.PROJECT_NAME_SLUG}} {{cookiecutter.PROJECT_NAME_SLUG}}/
+docker build -t agogosml -f agogosml/Dockerfile.agogosml agogosml/
+docker build -t input_reader -f input_reader/Dockerfile.input_reader input_reader/
+docker build -t output_writer -f output_writer/Dockerfile.output_writer output_writer/
+docker build -t sample_app -f <YOUR_APP_NAME>/Dockerfile.<YOUR_APP_NAME> <YOUR_APP_NAME>/
```
Comment (sayar) [inline agogosml_cli/cli/templates/{{cookiecutter.PROJECT_NAME_SLUG}}/dockerbuild.sh]: Why was this changed? cookiecutter was injecting the name of the project directly into the dockerbuild.sh file.

File: agogosml_cli/cli/templates/{{cookiecutter.PROJECT_NAME_SLUG}}/e2e/README.md [hunk 2]
```diff
@@ -7,52 +7,54 @@
...
 ```
 
-2. Run Docker Compose, which will also create a test generator application.
+2. Build the images for the base agogosml, input, output, and your application. We have provided a script for this named `dockerbuild.sh` in the root.
+
+3. Once the images are built, run the Docker Compose file `docker-compose-agogosml.yml`. This file spins up the three images we just built, and it will also create a test generator application. The testgen app will generate and send events to the chosen streaming service (Eventhub, Kafka) that will be picked up by the input-reader and start the pipeline. Once the data is run through the custom application, and sent to the messaging service, we will print them for you to validate the process.
 
 ```bash
-docker-compose -f docker-compose-agogosml.yml up --scale input-reader=2 -d
+docker-compose -f docker-compose-agogosml.yml up --scale input-reader=2
```
Comment (sayar) [inline agogosml_cli/cli/templates/{{cookiecutter.PROJECT_NAME_SLUG}}/e2e/README.md]: 👍

File: agogosml_cli/cli/templates/{{cookiecutter.PROJECT_NAME_SLUG}}/dockerbuild.sh (hunk 1 above)
Comment (margaretmeehan) [reply agogosml_cli/cli/templates/{{cookiecutter.PROJECT_NAME_SLUG}}/dockerbuild.sh]: ohhh silly me, thanks!

**Comment by margaretmeehan** [comment]
@sayar reminder to merge this in when possible. i addressed your comment
